src/api/java/com/teamacronymcoders/epos/api/locks/LockRegistry.java

```diff
@@ -0,0 +1,129 @@
+
+    public <KEY extends ILockKey> void registerLockType(@Nonnull ILockKeyCreator<KEY> creator) {
+        keyCreators.add(creator);
+        //Unlike the 1.12 system this has it so that each lock key has one registration for each type it can be created from
+        //Each one has to be checked when getting by type rather than knowing it "should" be good for that type already
+        //Does this have any negative performance impact
+    }
+
+    /**
+     * For use in reloading, and unit tests
+     */
+    public void clearLocks() {
+        locks.clear();
+        fuzzyLockInfo.clear();
+    }
+
+    public void addLockByKey(@Nonnull ILockKey key, @Nonnull List<IRequirement> requirements) {
+        if (key instanceof GenericLockKey || requirements.isEmpty()) {
+            //Don't allow having locks on the generic key that is just for when there isn't enough information about a fuzzy lock key
+            return;
+        }
+        locks.put(key, requirements);
+
+        if (key instanceof IFuzzyLockKey) {
+            IFuzzyLockKey fuzzy = (IFuzzyLockKey) key;
+            if (!fuzzy.isNotFuzzy()) {
+                //Store the fuzzy instance in a list for the specific item
+                fuzzyLockInfo.computeIfAbsent(fuzzy.getNotFuzzy(), k -> new HashSet<>()).add(fuzzy);
+            }
+        }
+    }
+
+    @Nonnull
+    public List<IRequirement> getRequirementsByKey(@Nonnull ILockKey key) {
+        if (key instanceof IFuzzyLockKey) {
+            return getFuzzyRequirements((IFuzzyLockKey) key);
+        }
+        return locks.getOrDefault(key, EMPTY_REQUIREMENTS);
+    }
+
+    @Nonnull
+    public List<IRequirement> getFuzzyRequirements(@Nonnull IFuzzyLockKey key) {
+        List<IRequirement> requirements = new ArrayList<>();
+        if (key.isNotFuzzy()) {
+            addRequirementsFromLock(requirements, key);
+        } else {
+            ILockKey baseLock = key.getNotFuzzy();
+            //Add the base lock's requirements
+            addRequirementsFromLock(requirements, baseLock);
+            Set<IFuzzyLockKey> fuzzyLookup = fuzzyLockInfo.get(baseLock);
+            if (fuzzyLookup != null) {
+                for (IFuzzyLockKey fuzzyLock : fuzzyLookup) {
+                    if (key.fuzzyEquals(fuzzyLock)) { //Build up the best match
+                        //fuzzy is the given object and has all info and fuzzyLock is the partial information
+                        addRequirementsFromLock(requirements, fuzzyLock);
+                    }
+                }
+            }
+        }
+        return requirements;
+    }
+
+    //Helper method to add a lock's requirements to the requirement list if it has them
+    private void addRequirementsFromLock(@Nonnull List<IRequirement> requirements, @Nonnull ILockKey key) {
+        List<IRequirement> reqs = locks.get(key);
+        if (reqs != null) {
+            //If there are locks for the key add them
+            requirements.addAll(reqs);
+        }
+    }
```

Comment:
We might want to put Helper methods in a separate helper class?

Reply:
It *may* make sense to move some of the methods from this to the LockCombiner class once I create one; though this method you are commenting on is `private`. It basically is used to reduce duplicated code as can be seen from the multiple calls to it; however, there is no reason things outside of this class should be using it so it would not make sense to put in a separate class.